src/subdomains/core/history/dto/transaction-refund.dto.ts

```diff
@@ -0,0 +1,16 @@
+import { ApiPropertyOptional } from '@nestjs/swagger';
+import { IsInt, IsNotEmpty, IsString, ValidateIf } from 'class-validator';
+
+export class TransactionRefundDto {
+  @ApiPropertyOptional()
+  @ValidateIf((b: TransactionRefundDto) => Boolean(b.refundUserId || !b.refundAddress))
+  @IsNotEmpty()
+  @IsInt()
+  refundUserId?: number;
```

Comment:
`refundUserId` is a bit pointless since the user IDs are not exposed via API, and externally we speak of addresses, not users.